Stop requiring event args to extend EventArgs (for .NET Standard)

blairconrad opened the pull request:
Fixes #361.

blairconrad commented:
The first commit just pays off some tech debt - we weren't running all the specs we could.

thomaslevesque commented on tests/FakeItEasy.Specs/EventRaisingSpecs.cs:
```diff
@@ -146,6 +158,21 @@ public void EventArguments()
+            "it should pass the event arguments"
+                .x(() => CapturedArgs1.Should().BeSameAs(this.referenceTypeEventArgs));
+        }
+#endif
```
These specs should be updated to GWT style

blairconrad replied:
Of course. Sorry/thanks!

thomaslevesque commented on tests/FakeItEasy.Specs/FakeItEasy.Specs.csproj:
```diff
@@ -24,7 +24,7 @@
     <Optimize>false</Optimize>
     <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>TRACE;DEBUG;FEATURE_SELF_INITIALIZED_FAKES</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;FEATURE_SELF_INITIALIZED_FAKES FEATURE_EVENT_ARGS_MUST_EXTEND_EVENTARGS</DefineConstants>
```
Nitpick: there's a inconsistency here... some symbols are separated by `';'`, others by `' '`

blairconrad replied:
I noticed, but when I looked at the help, it told me to separate by spaces, but the IDE added the semicolons after TRACE and DEBUG. I now see I was looking at the VS2012 help (why this is the default page when I hit F1 in VS2015, I could not tell you). 2015's help does say to use semicolons. Will fix.

thomaslevesque replied:
Not a big deal, since it works with either space or semicolons...

blairconrad commented:
* fixup:
    * semicolons, not spaces
    * in the same fixup, moved the new spec - I think it's better lower down
* GWT syntax
* a little more tidying in the specs - whitespace, removing a weird exception-catching method, that sort of thing

thomaslevesque commented on tests/FakeItEasy.Specs/EventRaisingSpecs.cs:
```diff
@@ -189,136 +204,167 @@ public void NullSenderAndEventArguments()
         {
-            "when raising custom event passing sender and arguments"
+            "Given an event of type EventHandler with custom event arguments type"
+                .See(() => nameof(Fake.GenericEvent));
```
It should be `CustomEvent` here, not `GenericEvent` (the description also needs to be changed)

blairconrad replied:
Whoops. So it should. Thanks! I think the test should appear below the new one as well.

blairconrad commented:
Thanks, @thomaslevesque. Rebase was more work than anticipated. I guess reordering members and touching half the lines in a file don't mix well. :)

While rebasing, I noticed we still had semicolons and spaces in FakeItEasy.csproj, so I fixed while I was at it. Too impatient to wait another round of review, so I squashed that right in. Here's the diff between what you reviewed and what I'm pushing:

```diff

λ  git diff origin/no-eventargs-netstd
diff --git a/src/FakeItEasy/FakeItEasy.csproj b/src/FakeItEasy/FakeItEasy.csproj
index 3e8ecb2..c241037 100644
--- a/src/FakeItEasy/FakeItEasy.csproj
+++ b/src/FakeItEasy/FakeItEasy.csproj
@@ -30,7 +30,7 @@
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>TRACE;DEBUG;FEATURE_BINARY_SERIALIZATION FEATURE_REFLECTION_GETASSEMBLIES FEATURE_SELF_INITIALIZED_FAKES FEATURE_EVENT_ARGS_MUST_EXTEND_EVENTARGS</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;FEATURE_BINARY_SERIALIZATION;FEATURE_REFLECTION_GETASSEMBLIES;FEATURE_SELF_INITIALIZED_FAKES;FEATURE_EVENT_ARGS_MUST_EXTEND_EVENTARGS</DefineConstants>^M
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <CodeAnalysisRuleSet>..\..\FakeItEasy.ruleset</CodeAnalysisRuleSet>
@@ -41,7 +41,7 @@
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
     <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE;FEATURE_BINARY_SERIALIZATION FEATURE_REFLECTION_GETASSEMBLIES FEATURE_SELF_INITIALIZED_FAKES FEATURE_EVENT_ARGS_MUST_EXTEND_EVENTARGS</DefineConstants>
+    <DefineConstants>TRACE;FEATURE_BINARY_SERIALIZATION;FEATURE_REFLECTION_GETASSEMBLIES;FEATURE_SELF_INITIALIZED_FAKES;FEATURE_EVENT_ARGS_MUST_EXTEND_EVENTARGS</DefineConstants>^M
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <DocumentationFile>Bin\Release\FakeItEasy.xml</DocumentationFile>
```

thomaslevesque commented:
Thanks @blairconrad!

blairconrad commented:
Ha, thank you.
